pkgs/build-support/build-dotnet-package/default.nix

```diff
@@ -0,0 +1,120 @@
+{ stdenv, lib, makeWrapper, pkgconfig, mono, dotnetbuildhelpers }:
+
+{ baseName
+, version
+, buildInputs ? []
+, placateNuget ? true
+, placatePaket ? true
+, patchFSharpTargets ? true
+, removeDuplicatedDlls ? true
+, xBuildFiles ? [ ]
+, xBuildFlags ? [ "/p:Configuration=Release" ]
+, outputFiles ? [ "bin/Release/*" ] # Wildcards allowed
+, dllFiles ? [] # Wildcards allowed
+, exeFiles ? [] # Wildcards NOT allowed
+, preConfigure ? ""
+, postConfigure ? ""
+, preBuild ? ""
+, postBuild ? ""
+, preInstall ? ""
+, postInstall ? ""
+, ... } @ attrs:
+  stdenv.mkDerivation ({
+    name = "${baseName}-${version}";
+
+    buildInputs = [
+      pkgconfig
+      mono
+      dotnetbuildhelpers
```

Comment:
Don't use this `dotnetbuildhelpers`, it's just one file so you could put it in this directory.